pkg/installation/install.go

```diff
@@ -76,7 +76,17 @@ func Install(p environment.Paths, plugin index.Plugin, forceDownloadFile string)
 	if err != nil {
 		return err
 	}
-	return install(plugin.Name, version, uri, bin, p, fos, forceDownloadFile)
+
+	glog.V(2).Infof("Store install receipt for plugin %s", plugin.Name)
+	if err = receipt.Store(plugin, p.PluginReceiptPath(plugin.Name)); err != nil {
```

Comment:
[P1] I think we actually should save the receipt after `install()`.

This is better because:

- actually installing a plugin (extract files, symlink binary) but then failing to store the receipt is okay (at most you'd be leaking some files).
- claiming the plugin is installed (i.e. receipt exists) while it's not installed is not okay.
- tens of things can go wrong in `install()`, much fewer things can go wrong in `receipt.Store()`, so makes sense to do it last.

currently your changes to Uninstall() get this right.

Reply:
Agreed.

Reply:
Let's move receipt saving after install() succeeds.